docs/spl-docs/evaluation-functions/comparison-conditional-functions.md

```diff
@@ -0,0 +1,177 @@
+- `<value>` argument can be a field name, string or numeric literal or an expression.
+- If all the values are `NULL`, it returns `NULL`.
+- One can use this function with eval and where commands.
+
+#### Example
+The following command checks for the user identity from fields username, login_id or email and returns the first value that is not `NULL`. Specifically, it checks in order username, login_id, email. If all the three fields have `NULL` values, it returns `Unknown`.
+```
+... | eval user=coalesce(username, login_id, email, "Unknown")
+```
+
+
+
+
+## searchmatch(\<search_str\>)
+This function returns `true` if the record matches for the search string `<search_str>`. Else returns `false`.
+
+#### Usage
+- `<search_str>` argument is a string containing sets of field value pairs like `"field1=value1 field2=value2"`.
+- searchmatch will search for value in the corresponding field. If all the searches for field value pairs are successful, the functions returns `true`, else returns `false`.
+- Values presented in the search string can have wildcard characters `*` or `?`. 
+- wildcard character `*` is used to match multiple characters.
+- wildcard character `?` is used to match a single character.
+- `<search_str>` could also have a single string literal. In this case, this string literal would be searched in all the available fields and if anyone matches the `<search_str>`, it returns `true`, else `false`.
+- One can use this function with eval and where commands.
+
+#### Example
+The following command searches for occurence of string "error" in the field error_msg and also searches for http error codes in range of 500 in http_status field. If both the searches are successful the function returns `true`, else will return `false`. Thus, when used in combination with `if` command, the field error_type gets the value `"Server error"` if the searchmatch is true else would be `"Unknown"`
+```
+... | eval error_type=if(searchmatch("error_msg=*error* http_status=5??"), 
+                        "Server Error", 
+                        "Unknown")
+```
+
+
+
+
+## if(\<predicate\>, \<true_value\>, \<false_value\>)
+This function evaluates the `<predicate>` expression and if the expression returns `true`, the `<true_value>` is returned, else `<false_value>` is returned.
+
+#### Usage
+- `<predicate>` argument is a boolean expression that evaluates to `true` or `false`.
+- `<true_value>` and `<false_value>` can be a field name, string or numeric literal or an expression.
+- One can use this function with eval and where commands.
+
+#### Example
+The following command evaluates the given predicate, if the txn_amount is greater than `10000` and country is not `USA` then high_risk is set to `Yes` else set to `No`.
+```
+... | eval high_risk=if(txn_amount > 10000 AND country != "USA", "Yes", "No")
+```
+
+
+## in(\<value\>, \<list\>)
+This function checks if the `<value>` is present in the `<list>`. If present it returns `true`, else returns `false`.
```

Comment:
nit: the description can be something like:
```
This function returns whether `<value>` is present in `<list>`.
```

fyi: `If present it returns true, else returns false.` doesn't quite make sense. A better version would be `If present it returns true; otherwise it returns false.` You're doing this several other places and should change them